superset-frontend/src/dashboard/components/gridComponents/Header.jsx

```diff
@@ -176,11 +176,7 @@ class Header extends React.PureComponent {
                   showTooltip={false}
                 />
                 {!editMode && (
-                  <AnchorLink
-                    anchorLinkId={component.id}
-                    filters={filters}
-                    showShortLinkButton
-                  />
+                  <AnchorLink id={component.id} dashboardId={dashboardId} />
```

Comment:
Adding `dashboardId` fixes the bug where the copy button doesn't copy the right URL for section headers in dashboards.